### ISSUE TITLE
Support for recursive types?

### ISSUE THREAD

**Issue by Guillaume-iov42** [andyglow/scala-jsonschema#92]
The README mentions support for recursive types. But with my simple example below, I get a `StackOverflowError` at compile time. Am I missing anything?

```
case class Foo(a: Bar, b: String)
case class Bar(c: Foo, d: String)

val recursiveSchema = Json.schema[Foo].asCirce(Draft04())
println(recursiveSchema.toString())
```

**Comment by Guillaume-iov42** [comment]
Ah, I'm stupid, I was using version `0.5.0` for some reason...
I can see recursive types have been added in `0.6.0`, and indeed it works fine!

**Comment by andyglow** [comment]
No worries. Recursive types support is a relatively new feature and only available starting from 0.6.x version
